src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -311,14 +323,75 @@ TEST_F(PolicyHandlerTest, OnPermissionsUpdated_method_With2Parameters) {
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
   // Act
   Permissions perms;
   policy_handler_.OnPermissionsUpdated(kPolicyAppId_, perms);
 }
 
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_TwoParams_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(_, _, _)).Times(0);
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions);
+}
+
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions, "HMI_FULL");
+}
+
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_HmiLevelInvalidEnum_UNSUCCESS) {
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillRepeatedly(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, app_id()).WillRepeatedly(Return(kAppId1_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  const HMILevel default_hmi("INVALID_ENUM");
+  EXPECT_CALL(mock_message_helper_, StringToHMILevel(default_hmi))
+      .WillOnce(Return(mobile_apis::HMILevel::INVALID_ENUM));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
+  EXPECT_CALL(app_manager_, state_controller()).Times(0);
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions, default_hmi);
+}
+
+TEST_F(PolicyHandlerTest,
+       OnPermissionsUpdated_HmiLevelEqualsToCurrentHmiLevel_UNSUCCESS) {
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillRepeatedly(Return(mock_app_));
+  EXPECT_CALL(*mock_app_, app_id()).WillRepeatedly(Return(kAppId1_));
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+
+  const HMILevel default_hmi("HMI_NONE");
+  EXPECT_CALL(mock_message_helper_, StringToHMILevel(default_hmi))
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_NONE));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
+  EXPECT_CALL(app_manager_, state_controller()).Times(0);
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions, "HMI_NONE");
+}
```

Comment:
@okozlovlux here and test above I see the same logic maybe need to union these tests and call them with different parameters